syftlib/src/main/java/org/openmined/syft/monitor/DeviceMonitor.kt

```diff
@@ -0,0 +1,25 @@
+package org.openmined.syft.monitor
+
+import io.reactivex.processors.PublishProcessor
+import org.openmined.syft.domain.SyftConfiguration
+import org.openmined.syft.monitor.battery.BatteryStatusRepository
+import org.openmined.syft.monitor.network.NetworkStatusRepository
+
+@ExperimentalUnsignedTypes
+class DeviceMonitor(
+    syftConfig: SyftConfiguration
+) {
+
+    private val networkStatusRepository =
```

Comment:
These could be injected in the constructor.

Reply:
Yeah will open up a static function to create the instance using injection